src/test/java/service/impl/FileReaderImplTest.java

```diff
@@ -0,0 +1,59 @@
+package service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static service.impl.DataConverterImplTest.createTransaction;
+
+import java.util.Arrays;
+import java.util.List;
+import model.FruitTransaction;
+import org.junit.Before;
+import org.junit.Test;
+import service.DataConverter;
+
+public class FileReaderImplTest {
+    private static final String VALID_FILE = "src/test/resources/test.csv";
+    private static final String EMPTY_FILE = "src/test/resources/empty.csv";
+    private static final String NON_EXISTENT_FILE = "src/test/resources/nonExistentFile.csv";
+
+    private FileReaderImpl fileReader;
+
+    @Before
+    public void setUp() {
+        fileReader = new FileReaderImpl();
+    }
+
+    @Test
+    public void readValidFile_ok() {
+        List<String> lines = fileReader.read(VALID_FILE.toString());
+
+        List<FruitTransaction> expected = Arrays.asList(
+                createTransaction("b", "banana",20),
```

Comment:
Don’t call methods from other test classes. Use the FruitTransaction constructor or move this method to the util class